mempool/mempool
# Add 'db-less' mining pool tagging support

Fixes https://github.com/mempool/mempool/issues/1821

This PR fixes a regression introduced when we refactored away the mining pool tagging logic from the frontend. Today we rely on the backend to tag mining pools, but the problem is that it also assume the database is available.

With this PR, RAM memory is used as a temporary storage for mining pools when the database is disabled.

### Testing

* Clear your disk cache, to make sure we process at least a few blocks for testing this new code
* Set `DATABASE.ENABLED: false` in `mempool-config.json`
* Set `MINING_DASHBOARD: false` in `mempool-frontend-config.json`
* Start the node backend
* Browse blocks and confirm that they have proper mining pool tag, even the older one

Also, there should be no changes at all from `master` with `DATABASE.ENABLED: true` and `MINING_DASHBOARD: true`
